adhere to Debian version comparison guidelines

camilajenny opened the pull request:
see my comments on https://github.com/donadigo/eddy/pull/140

and my tests https://gist.github.com/camilajenny/2059145ad34e4e5c3adbff47ccca7657

and it adheres to the official guidelines https://www.debian.org/doc/debian-policy/ch-controlfields.html#version

Please review.

benprew commented:
@camilajenny can you include your test script as part of this PR? Having a test script that can be rerun later in case someone wants to change the code would be useful.

You can create the test/ directory in the root and put the files there. See geary as an example:

https://gitlab.gnome.org/GNOME/geary/-/tree/mainline

Thanks

camilajenny commented:
@benprew thank you for your suggestion. I added the ValaUnit to the subprojects and I created a unit test, and added the test script instructions to the unit test. It's easily extensible now, in case we need more unit tests. I did it like in geary, and ValaUnit is LGPL so we can use and modify it here without any problem. I also fixed the upgrade / downgrade flag which I broke in previous commits.

Please review again.
